src/mpyl/stages/discovery.py

```diff
@@ -87,15 +87,55 @@ def is_dependency_touched(
     return False
 
 
-def is_output_cached(output: Optional[Output], cache_key: str) -> bool:
-    if (
-        output is None
-        or not output.success
-        or output.produced_artifact is None
-        or not output.produced_artifact.hash
-    ):
-        return False
-    return output.produced_artifact.hash == cache_key
+def is_project_cached_for_stage(
+    logger: logging.Logger,
+    project: str,
+    stage: str,
+    output: Optional[Output],
+    cache_key: str,
+) -> bool:
+    if stage == deploy.STAGE_NAME:
+        logger.debug(
+            f"Project {project} will execute stage {stage} again because this stage is never cached"
+        )
+        cached = False
```

Comment:
Just put this is as the default value instead of setting it in every `if`?

Reply:
My preference would be to just do:
```python
if something:
  return False
elif something else:
  return False
else:
  return True
```

But apparently it's against the project style guide 🤷